src/systemtask/SystemTask.cpp

```diff
@@ -370,22 +365,54 @@ void SystemTask::UpdateMotion() {
   if (isSleeping)
     twiMaster.Wakeup();
 
-  if (stepCounterMustBeReset) {
+  if (isSleeping)
+    twiMaster.Sleep();
+
+  auto motionValues = motionSensor.Process();
+  stepCounterUnsavedCount += motionValues.steps;
+  motionSensor.ResetStepCounter();
+
+  // As the UpdateMotion is triggered every 100ms or so we will not save
+  // the value every tick. With the code below we will save and restore
+  // values every 10th second. (100 * 100ms = 10s)
+  if (stepCounterSaveTimer > 100 && stepCounterUnsavedCount > 0) {
     motionSensor.ResetStepCounter();
-    stepCounterMustBeReset = false;
+    stepCounterSavedCount = SavingSteps(stepCounterUnsavedCount);
+    stepCounterSaveTimer = 0;
+    stepCounterUnsavedCount = 0;
   }
 
-  auto motionValues = motionSensor.Process();
-  if (isSleeping)
-    twiMaster.Sleep();
+  stepCounterSaveTimer++;
 
   motionController.IsSensorOk(motionSensor.IsOk());
-  motionController.Update(motionValues.x, motionValues.y, motionValues.z, motionValues.steps);
+  motionController.Update(motionValues.x, motionValues.y, motionValues.z, stepCounterSavedCount + stepCounterUnsavedCount);
+
   if (motionController.ShouldWakeUp(isSleeping)) {
     GoToRunning();
   }
 }
 
+uint32_t SystemTask::SavingSteps(uint32_t steps) {
+  lfs_file_t stepsFile;
+  uint32_t oldSteps;
+
+  char buff[40];
+  snprintf(buff, sizeof(buff), "/steps_%.4d%.2d%.2d.dat", dateTimeController.Year(), static_cast<u_int8_t>(dateTimeController.Month()), dateTimeController.Day());
```

Comment:
You should use the standard defined type `[std::]uint8_t` in the cast rather than `u_int8_t`.

Reply:
Hi @jonvmey 

That code was copied from the settings.cpp file where we use the filesystem. I'll test to change this and see if it works later.

Best regards
Daniel